src/System.Device.Gpio.Tests/GpioControllerTestBase.cs

```diff
@@ -0,0 +1,159 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Threading;
+using Xunit;
+[assembly: CollectionBehavior(DisableTestParallelization = true)]
+
+namespace System.Device.Gpio.Tests
+{
+    public abstract class GpioControllerTestBase
+    {
+        private const int LedPin = 18;
+        private const int OutputPin = 16;
+        private const int InputPin = 12;
+
+        [Fact]
+        public void ControllerCanTurnOnLEDs()
+        {
+            using (GpioController controller = new GpioController(GetTestNumberingScheme(), GetTestDriver()))
+            {
+                controller.OpenPin(LedPin, PinMode.Output);
+                Thread.Sleep(1_000);
+                controller.Write(LedPin, PinValue.High);
+                Thread.Sleep(TimeSpan.FromSeconds(1));
+                controller.Write(LedPin, PinValue.Low);
+            }
+        }
+
+        [Fact]
+        public void PinValueReturnsToLowAfterDispose()
```

Comment:
NIT This method name seems not matching with test, maybe I misunderstood ...

Reply:
It does do that, In the first using block we will set the value to high, and then the pin will be disposed when the controller is disposed. Then we initialize a new controller so we validate the the pin value is set to low again.

Reply:
ah got it, thank you!